### PR TITLE
Update the exports in generated *.js and *.d.ts files

### PR DESCRIPTION
**Pull request by daschult**
The main change in this pull request is in CodeModelJs.cs. In that file, I extended the *.js file exports from:
```javascript
module.exports = AutoRestParameterGroupingTestService;
```
to
```javascript
module.exports = AutoRestReportServiceForAzure;
module.exports['default'] = AutoRestReportServiceForAzure;
module.exports.AutoRestReportServiceForAzure = AutoRestReportServiceForAzure;
module.exports.AutoRestReportServiceForAzureModels = models;
```
This allows developers to import our Node.js SDKs using either of the following methods:
```typescript
// Typescript
import AutoRestSwaggerBATArrayService from '../Expected/AcceptanceTests/BodyArray/autoRestSwaggerBATArrayService';
import { AutoRestSwaggerBATdictionaryService, AutoRestSwaggerBATdictionaryServiceModels } from '../Expected/AcceptanceTests/BodyDictionary/autoRestSwaggerBATdictionaryService';
```
```javascript
// JavaScript
var azureCompositeClient = require('../Expected/AcceptanceTests/AzureCompositeModelClient/azureCompositeModel');
```
```

### CODE REVIEW

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Comment by daschult** [comment]
This PR resolves issue https://github.com/Azure/autorest.nodejs/issues/23.

**Comment by daschult** [comment]
Correct.

**Comment by olydis** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.4.2*

## ❌️ General: 0%

no tests were run for this category


## ✔️ Azure: 100%

**Comment by olydis** [comment]
# 🤖 AutoRest automatic feature coverage report 🤖
*feature set version 2.4.2*

## ❌️ General: 0%

no tests were run for this category


## ✔️ Azure: 100%

File: src/vanilla/Model/CodeModelJs.cs [hunk 1]
```diff
@@ -230,9 +230,10 @@ public bool ContainsTimeSpan
         public string ConstructServiceClientJSExports()
         {
             StringBuilder builder = new StringBuilder();
-            builder.AppendLine($"exports.{Name} = {Name};");
-            builder.AppendLine($"exports.{GetServiceModelsName(Name)} = models;");
-            builder.AppendLine($"exports[\"default\"] = {Name};");
+            builder.AppendLine($"module.exports = {Name};");
+            builder.AppendLine($"module.exports['default'] = {Name};");
```
Comment (amarzavery) [inline src/vanilla/Model/CodeModelJs.cs]: @daschult - Out of curiosity: Was there a reason to use module.exports['default'] and not module.exports.default ?

**Comment by amarzavery** [comment]
@daschult @RikkiGibson - I have tagged 2.1.32 as latest so whenever you do autorest refresh and then execute autorest --nodejs .... then you should get this as the latest version by default.

File: src/vanilla/Model/CodeModelJs.cs (hunk 1 above)
Comment (daschult) [reply src/vanilla/Model/CodeModelJs.cs]: That was similar to the code that Typescript generated in some of my handwritten tests, so I went with that. I don't think there was any other reason.

Comment (RikkiGibson) [reply src/vanilla/Model/CodeModelJs.cs]: Maybe it's because default is a keyword in es6, so the expression `obj.default` isn't allowed.

**Pull request merged**